### ISSUE TITLE
Feature Request: auto detect source vs target translation files

### ISSUE THREAD

**Issue by sventendo** [onigunn/intellij.xliff#4]
When selecting a translation file, we could check for a ISO prefix to know if the translation needs this:

(locallang.xliff)
```
            <trans-unit id="my_translation">
                <source>My original text</source>
            </trans-unit>
```

or that:
(de.locallang.xliff)
```
            <trans-unit id="my_translation">
                <target>Meine Übersetzung</target>
            </trans-unit>
```

**Comment by onigunn** [comment]
Sorry, I don't grasp what you want to achieve. Could you dig a little bit deeper?

**Comment by sventendo** [comment]
The original "translation" file is not a translation at all, rather a static string replacement tool. The file is per default "locallang.xliff" - with no prefix.

Now if I want to actually add a string to a translation file and not the original language file, then it needs to be a prefixed file, e.g. "de.locallang.xliff" and it needs to have at least a <source> tag. The <target> tag is optional.

Ideally every translation in "de.locallang.xliff" would have a default value in "locallang.xliff" but in the real world, with projects that are not multi-language, you still want the proper translation file but you often would skip the default file.

So the use case is this:
- select a string in a template
- extract to language file
- if it's a target language file, the plugin should generate a <target> tag
- if it's the default language file, the plugin should generate a <source> tag

How does the plugin know it's a target language file? Check for a prefix.

That's the idea. I believe this feature would be very helpful to non-English projects.

Does this make more sense now?

**Comment by onigunn** [comment]
Yep. Now I got you. Your idea isn't bad at all but would conflict with a WIP feature I'm working at the moment with. Target is to provide a custom editor UI to easily add/update missing translations based on the locallang.xlf. It will look and work like the java resource bundle editor. There will be no need to select a prefixed file then. You will only interact with the locallang.xlf.

Feature is currently WIP.

**Comment by mai-space** [comment]
Also I would like to suggest an additional Enhancement to this feature.
Maybe select the Text in the Editor and Extract it always to the default locallang.
Than get all translations of that file.
Translate with Deepl API to the target language.
I have used the Deepl API in VSC in an Extension and have dreamt of that feature in PHPStorm.
Maybe, if I got time for this, I will fork and create a PR...
